sdk/src/Core/Amazon.Runtime/Internal/Transform/UnmarshallerContext.cs

```diff
@@ -243,6 +244,14 @@ private static bool TestExpression(string expression, int startingStackDepth, st
         /// </returns>
         public abstract bool Read();
 
+        /// <summary>
+        /// Reads the next node in the document and updates the context accordingly.
+        /// This is specifically used for JSON protocols which require a reader to be passed in.
+        /// </summary>
+        /// <param name="reader">The Utf8JsonReader</param>
+        /// <returns></returns>
+        public abstract bool Read(ref StreamingUtf8JsonReader reader);
```

Comment:
Good point, I removed both the abstract `Read()` with no parameters and abstract `Read(ref streamingUtf8JsonReader reader)` and moved each one to its unmarshaller context class. I also moved `ReadAtDepth(int targetDepth)`  since the Json unmarshaller context's `ReadAtDepth` accepts an additional argument anyways.